### ISSUE TITLE
在推理生成和答案推理两个阶段的训练中都出现了'loss': 0.0, 'grad_norm': nan

### ISSUE THREAD

**Issue by WRJ3431** [chengtan9907/mc-cot#2]
使用一张4090显卡在windows上进行训练，bs=2，具体命令如下：
python main.py --model ./models/unifiedqa-t5-base --user_msg rationale --img_type detr --bs 2 --eval_bs 4 --eval_acc 10 --output_len 512 --final_eval --prompt_format QCM-LE --epoch 50 --vot_num 5 --alpha 0.5 --output_dir ./results/base_train_from_scratch
终端信息：
(VQA) D:\a_github_projects\mc-cot-master>python main.py --model ./models/unifiedqa-t5-base --user_msg rationale --img_type detr --bs 2 --eval_bs 4 --eval_acc 10 --output_len 512 --final_eval --prompt_format QCM-LE --epoch 50 --vot_num 5 --alpha 0.5 --output_dir ./results/base_train_from_scratch
args Namespace(data_root='./data', output_dir='./results/base_train_from_scratch', model='./models/unifiedqa-t5-base', options=['A', 'B', 'C', 'D', 'E'], epoch=50, lr=5e-05, bs=2, input_len=512, output_len=512, vot_num=5, alpha=0.5, eval_bs=4, eval_acc=10, train_split='train', val_split='val', test_split='test', resume_from_checkpoint=None, use_generate=False, final_eval=True, user_msg='rationale', img_type='detr', eval_le=None, test_le=None, evaluate_dir=None, caption_file='data/captions.json', use_caption=False, prompt_format='QCM-LE', seed=42)
====Input Arguments====
{
  "data_root": "./data",
  "output_dir": "./results/base_train_from_scratch",
  "model": "./models/unifiedqa-t5-base",
  "options": [
    "A",
    "B",
    "C",
    "D",
    "E"
  ],
  "epoch": 50,
  "lr": 5e-05,
  "bs": 2,
  "input_len": 512,
  "output_len": 512,
  "vot_num": 5,
  "alpha": 0.5,
  "eval_bs": 4,
  "eval_acc": 10,
  "train_split": "train",
  "val_split": "val",
  "test_split": "test",
  "resume_from_checkpoint": null,
  "use_generate": false,
  "final_eval": true,
  "user_msg": "rationale",
  "img_type": "detr",
  "eval_le": null,
  "test_le": null,
  "evaluate_dir": null,
  "caption_file": "data/captions.json",
  "use_caption": false,
  "prompt_format": "QCM-LE",
  "seed": 42
}
img_features size:  (11208, 100, 256)
number of train problems: 12726

number of val problems: 4241

number of test problems: 4241

You are using the default legacy behaviour of the <class 'transformers.models.t5.tokenization_t5.T5Tokenizer'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565
[19:44:20] [Model]: Loading ./models/unifiedqa-t5-base...                                                                                                                                                                   main.py:95
           [Data]: Reading data...                                                                                                                                                                                          main.py:96
           [Model]: Loading ./models/unifiedqa-t5-base...                                                                                                                                                                  main.py:120
Some weights of T5ForMultimodalGenerationMCCoT were not initialized from the model checkpoint at ./models/unifiedqa-t5-base and are newly initialized: ['gate_dense.bias', 'gate_dense.weight', 'image_dense.bias', 'image_dense.weight', 'mha_layer.in_proj_bias', 'mha_layer.in_proj_weight', 'mha_layer.out_proj.bias', 'mha_layer.out_proj.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
  0%|                                                                                                                                                                                                     | 0/318150 [00:00<?, ?it/s]Passing a tuple of `past_key_values` is deprecated and will be removed in Transformers v4.48.0. You should pass an instance of `EncoderDecoderCache` instead, e.g. `past_key_values=EncoderDecoderCache.from_legacy_cache(past_key_values)`.
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.9968568285399966e-05, 'epoch': 0.03}
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.993713657079994e-05, 'epoch': 0.06}
  0%|▎                                                                                                                                                                                       | 558/318150 [03:20<29:35:51,  2.98it/s]T
  请问这种情况该怎么解决？
